multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/deadletterqueue/MultiTenantDeadLetterQueue.java

```diff
@@ -0,0 +1,406 @@
+        } else {
+            logger.info("No tenant found for current thread. Returning total size of all tenants queues.");
+            return tenants.stream().mapToLong(tenant -> fetchFromTenantSegment(tenant, SequencedDeadLetterQueue::size))
+                          .sum();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long sequenceSize(Object sequenceIdentifier) {
+        TenantDescriptor currentTenant = TenantWrappedTransactionManager.getCurrentTenant();
+        if (currentTenant != null) {
+            return fetchFromTenantSegment(currentTenant, seg -> seg.sequenceSize(sequenceIdentifier));
+        } else {
+            logger.info("No tenant found for current thread. Returning total size of sequences.");
+            return tenants.stream().filter(tenant -> fetchFromTenantSegment(tenant,
+                                                                            seg -> seg.contains(sequenceIdentifier)))
+                          .findFirst().map(tenant -> fetchFromTenantSegment(tenant,
+                                                                            seg -> seg.sequenceSize(sequenceIdentifier)))
+                          .orElse(0L);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long amountOfSequences() {
+        TenantDescriptor currentTenant = TenantWrappedTransactionManager.getCurrentTenant();
+        if (currentTenant != null) {
+            return fetchFromTenantSegment(currentTenant, SequencedDeadLetterQueue::amountOfSequences);
+        } else {
+            logger.info("No tenant found for current thread. Returning total amount of sequences.");
+            return tenants.stream().mapToLong(tenant -> fetchFromTenantSegment(tenant,
+                                                                               SequencedDeadLetterQueue::amountOfSequences))
+                          .sum();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean process(Predicate<DeadLetter<? extends M>> sequenceFilter,
+                           Function<DeadLetter<? extends M>, EnqueueDecision<M>> processingTask) {
+        TenantDescriptor currentTenant = TenantWrappedTransactionManager.getCurrentTenant();
+        if (currentTenant != null) {
+            return fetchFromTenantSegment(currentTenant, seg -> seg.process(sequenceFilter, processingTask));
+        } else {
+            logger.info("No tenant found for current thread. Processing all tenants queues.");
+            return tenants.stream().map(tenant -> fetchFromTenantSegment(tenant,
+                                                                         seg -> seg.process(sequenceFilter,
+                                                                                            processingTask))).reduce(
+                    false,
+                    (a, b) -> a || b);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
I would specify that this process operation will return `true` for *any* sequence it processes whenever the fallback approach is taken.
This needs to be described as it differs from the original JavaDoc.